src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+            AtEmptyException {
+        dividerPosition = userInput.indexOf("/at");
+        eventDescription = userInput.substring(START_OF_STRING, dividerPosition);
+        eventDescription = eventDescription.replace("event", "");
+        eventDescription = eventDescription.trim();
+        charAfterDividerPosition = dividerPosition + 1;
+        at = userInput.substring(charAfterDividerPosition);
+        at = at.replace("at", "");
+        at = at.trim();
+        if (at.equals("")) {
+            throw new AtEmptyException();
+        }
+    }
+
+    private static void performAddTodo(String todoTask) {
+        System.out.println(HORIZONTAL_LINE);
+        taskArrayList.add(new Todo(todoTask));
+        System.out.println("Got it. I've added this task:");
+        System.out.println(INDENT + taskArrayList.get(Task.taskCount - 1));
+        System.out.println("Now you have " + Task.taskCount + " tasks in the list");
+    }
+
+    private static void performAddDeadline(String description, String by) {
+        System.out.println(HORIZONTAL_LINE);
+        taskArrayList.add(new Deadline(description, by));
+        System.out.println("Got it. I've added this task:");
+        System.out.println(INDENT + taskArrayList.get(Task.taskCount - 1));
+        System.out.println("Now you have " + Task.taskCount + " tasks in the list");
+    }
+
+    private static void performAddEvent(String description, String at) {
+        System.out.println(HORIZONTAL_LINE);
+        taskArrayList.add(new Event(description, at));
+        System.out.println("Got it. I've added this task:");
+        System.out.println(INDENT + taskArrayList.get(Task.taskCount - 1));
+        System.out.println("Now you have " + Task.taskCount + " tasks in the list");
+    }
+
+    public static void exitProgram() {
+        printGreeting("Bye. Hope to see you again soon!", HORIZONTAL_LINE);
+        System.exit(0);
+    }
+
+    private static void saveFile() {
+        try {
+            createFolder();
+            createFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void createFolder() {
+        File folder = new File(FOLDERPATH);
+        if (folder.mkdir()) {
+            System.out.println("Folder created: " + folder.getName());
+        }
+    }
+
+    private static void createFile() throws IOException {
+        File f = new File(FILEPATH);
+        if (f.createNewFile()) {
+            System.out.println("File created: " + f.getName());
+        }
+        writeToFile();
+    }
+
+    private static void writeToFile() throws IOException {
+        FileWriter fw = new FileWriter(FILEPATH);
+        for (Task t : taskArrayList){
+            fw.write(t.taskNum + "." + t + System.lineSeparator());
+        }
+        fw.close();
+        System.out.println("File \"duke.txt\" updated");
+    }
```

Comment:
One suggestion is, instead of writing all these methods in Duke.java, you can extract out these methods that deals with files into a class (such as StorageManager.java) to make use of OOP to modularise your code and improve code readability.